delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ElasticSearchViewValue.scala

```diff
@@ -49,7 +49,7 @@ object ElasticSearchViewValue {
       includeMetadata: Boolean = false,
       includeDeprecated: Boolean = false,
       mapping: JsonObject,
-      settings: Option[JsonObject] = None,
+      settings: JsonObject,
```

Comment:
Without a default value? I think we use the `ElasticSearchViewValue` to generate the `deriveConfigJsonLdDecoder`. If there is no default value it means the client to always provide the `settings` field. Or am I missing something?

Reply:
I see you added a `applyDefaultSettings` method to hack around this. However you shouldn't do it this way. Notice that `ElasticSearchViewValue` is just an intermediate representation used to easily transform a client json-ld to the target case class `ElasticSearchView`.

Here, what you should do instead is keep `settings: Option[JsonObject] = None` and then just have a method:
```scala
def settingsOrDefault: JsonObject = ???
```

Reply:
That's how we did it for [StorageFields](https://github.com/BlueBrain/nexus/blob/master/delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageFields.scala#L86) which is sort of the same use case

Reply:
`ElasticSearchViewValue` is not an intermediate representation like `StorageFields`, that is why I proceeded this way as `ElasticSearchViewValue` is part of ES events.

Reply:
Still, it is not what one interact with from the SDK. You interact with `ElasticSearchView`. But I see why would be useful to have it on the event...

Reply:
I also agree that we should have a consistent approach between resolvers/storages/views.

But it goes beyond this PR no ?

Reply:
I think for now it is better the other approach, where you keep settings as Option in `ElasticSearchViewValue` since it is the safest.

Reply:
I'm happy with both options. I do see your point @umbreak, but I don't see any large implications going one way or another.